packages/drivers/odsp-driver/src/fetchWithRetry.ts

```diff
@@ -255,8 +256,8 @@ async function logFetchResponse(
 ) {
     if (logger !== undefined) {
         const additionalProps = getAdditionalProps && (await getAdditionalProps(response, isFinalAttempt));
-        logger.sendTelemetryEvent({
-            subCategory: "Request",
+        const childLogger = ChildLogger.create(logger, "Request");
```

Comment:
While there is nothing wrong with this approach, it feels a bit wasteful (in terms of creating logger for trivial operation), as really all you need is to do `Request:${nameForLogging}` concatenation for name, I think. If so, would it be more useful to expose a static API from logger to join sub-category & name?

Reply:
I'm looking at prepareEvent() event, it just concatenates everything into one string, so I assume approach is viable

Reply:
Why cann't we just set the name as Request:${nameForLogging} ? What does the subcategory do?

Reply:
We absolutely can and it will work. The only con of that - ":" is not really public knowledge in the code, i.e. if in future logger would change to use other symbol, we would need to chase code like that and change it. Abstraction allows us to have knowledge in one place.

Reply:
my thought here is we should think about what this telemetry is for, and how you think you'd want to query it. encoding in the name won't make it very easy to query for these entires specifically, as you'll have to parse the name, which is not performant, or robust. also should this be a perf event rather than a generic event?

Reply:
to answer your question above, subcategory create a new table which isn't necessary here

Reply:
ChildLogger is pretty lightweight and I think is the right abstraction rather than encoding special characters in event names line-by-line.

Reply:
That said, in this case I don't think the child logger is necessary here anyway.  How about we craft the event like this:

```
{
  eventName: "odspFetchResponse",
  requestName: nameForLogging,
  ...
```

Reply:
I think this code was taken FFX repo. The suggestion should good to me.